### PR TITLE
Publish SNAPSHOT builds to sonatype on pushes to develop

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What does this PR do?

Add 'SNAPSHOT' as a version tag, and default new versions to it. Change the build so that `develop` triggers all `publish` stages.

### Motivation

The cross platform frameworks would like to work from SNAPSHOTS in order to speed up development.

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -317,6 +326,7 @@ publish:release-timber:
   tags: [ "runner:main" ]
   only:
     - tags
+    - develop
```
Comment (mariusc83) [inline .gitlab-ci.yml:329]: if you add this here, doesn't this mean that this task will always be performed when develop builds ? That means we are going to push snapshots every time we merge something in develop, is that correct ?

Comment (xgouchet) [reply .gitlab-ci.yml:329]: Yes that's the goal

**Review by xgouchet — APPROVED** [review]

File: buildSrc/src/main/kotlin/com/datadog/gradle/config/AndroidConfig.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ object AndroidConfig {
     const val MIN_SDK_FOR_COMPOSE = 21
     const val BUILD_TOOLS_VERSION = "31.0.0"
 
-    val VERSION = Version(1, 15, 0, Version.Type.Dev)
+    val VERSION = Version(1, 15, 0, Version.Type.Snapshot)
```
Comment (0xnm) [inline buildSrc/src/main/kotlin/com/datadog/gradle/config/AndroidConfig.kt:20]: is it enough to publish the snapshot on Sonatype? will it skip staging/approval step if version has `-SNAPSHOT` suffix? Or will it be some other PR?

**Review by mariusc83 — APPROVED** [review]

File: buildSrc/src/main/kotlin/com/datadog/gradle/config/AndroidConfig.kt (hunk 2 above)
Comment (xgouchet) [reply buildSrc/src/main/kotlin/com/datadog/gradle/config/AndroidConfig.kt:20]: Sonatype will automatically skip staging based on the `-SNAPSHOT` suffix

**Review by xgouchet — APPROVED** [review]

**Pull request merged**